src/interfaces/Stats.hpp

```diff
@@ -0,0 +1,171 @@
+#pragma once
+
+/// @brief Struct que representa las stats ofensivas
+typedef struct {
+  float physicalDamage;
+  float magicDamage;
+  float attackSpeed;
+  float criticalChance;
+  float criticalDamage;
+  float lifeSteal;
+} OffensiveStats;
+
+/// @brief Struct que representa las stats defensivas
+typedef struct {
+  float health;
+  float healthMax;
+  float movementSpeed;
+  float agility;
+  float armor;
+  float resistance;
+  float healthRegeneration;
+} DefensiveStats;
+
+// Clase que almacena estatisticas
+class Stats {
+protected:
+  /// @brief Puntos de vida de la entidad
+  float health;
+
+  /// @brief Puntos de vida maximos de la entidad
+  float healthMax;
+
+  /// @brief Cuantos pixeles se mueve por frame
+  float movementSpeed;
+
+  /// @brief Probabilidad entre 0 - 30 de esquivar un golpe
+  float agility;
+
+  /// @brief Cuantos ataques por segundo se realizan
+  float attackSpeed;
+
+  /// @brief Puntos de daño de tipo físico que se infligen al atacar
+  float physicalDamage;
+
+  /// @brief Puntos de daño de tipo mágico que se infligen al atacar
+  float magicDamage;
+
+  /// @brief Probabilidad entre 0 - 30 que reduce el daño recibido de tipo
+  /// físico
+  float armor;
+
+  /// @brief Probabilidad entre 0 - 30 que reduce el daño recibido de tipo
+  /// mágico
+  float resistance;
+
+  /// @brief Probabilidad entre 0 - 30 de aplicar más daño físico / mágico
+  float criticalChance;
+
+  /// @brief Multiplicador entre 1 - 10 al daño infligido
+  float criticalDamage;
+
+  /// @brief Porcentaje entre 0 - 50 de curación del daño al atacar
+  float lifeSteal;
+
+  /// @brief Cantidad de puntos de vida que te curas cada segundo
+  float healthRegeneration;
```

Comment:
```suggestion
OffensiveStats offensiveStats;
DefensiveStats defensiveStats;
```

Ya que has creado 2 structs para los tipos de estadísticas usalos para definir las mismas y así tener mayor consistencia y cohesión.

Reply:
Esto no me convence porque al final entonces no es una clase stats. Los struct son auxiliares para manejar los datos fuera de la clase y darme la posibilidad de devolver varias stats como si fuera un array pero creo que se deberia de mantener las variables independientes de los structs